src/java.base/share/classes/jdk/internal/classfile/impl/AbstractInstruction.java

```diff
@@ -705,6 +707,59 @@ public String toString() {
 
     }
 
+    public static final class BoundJsrInstruction
+            extends BoundInstruction implements DiscontinuedInstruction.JsrInstruction {
+
+        public BoundJsrInstruction(Opcode op, CodeImpl code, int pos) {
+            super(op, op.sizeIfFixed(), code, pos);
+        }
+
+        @Override
+        public Label target() {
+            return offsetToLabel(branchByteOffset());
+        }
+
+        public int branchByteOffset() {
+            return size == 3
+                   ? (int) (short) code.classReader.readU2(pos + 1)
```

Comment:
```suggestion
                   ? code.classReader.readS2(pos + 1)
```
the assembly of this offset is the same as the assembly of the increment amount in wide iinc.